src/pages/ServerRoleSummary/NicMappingTab.js

```diff
@@ -68,6 +69,17 @@ class NicMappingTab extends Component {
     }
   }
 
+  getNicMappingsInUse = (internalModel) => {
+    let nicmappings =
+      internalModel['internal']['servers']?.filter(server => !isEmpty(server['ardana_ansible_host']))
+        .map(server => server['nic_map']);
+    // remove the duplicates
+    nicmappings =
+      [...new Set(nicmappings.map(nMapping => JSON.stringify(nMapping)))]
```

Comment:
I'm having difficulty envisioning why we need to `JSON.stringify` and then undo that again on the next line with `JSON.parse`. And if there is a good reason, please comment why and what this is doing.

Reply:
After discussing via rocket chat, it appears that this is trying to find unique objects in a list, and each object has many properties and is fairly deeply nested. This is converting each object to a string and then looking for unique strings. That is fundamentally flawed in JavaScript because the order of entries in an object is not specified, and two objects containing the same contents may be converted to different strings.
I don't quite know the context for what's going on here, but NIC mapping names are unique in the model, so shouldn't it be enough to just compare the name fields?

Reply:
@GarySmith  thanks for the review. Will update.